### DISCUSSION TITLE
double width chars

### DISCUSSION THREAD

**Discussion by emdete** [vlang/v#18609] [Ideas]
i havent found how term.ui supports double width chars (like 🙂). it would be quite helpful as modern terminals support this (and to be earnest the good old vt340 did so 50 years ago already).

**Comment by Delta456** [comment]
Can you give an example where it doesn't works for you?

**Comment by emdete** [comment]
the API just doesn't tell the width of a char on the  screen, if i use draw_text it would be helpful to have the width actually used on the screen to calculate how much text fits in the line (an additional function to get the width beforehand would be even better)
